rfcs/72-editor.md

```diff
@@ -0,0 +1,508 @@
+for modularity. An interactive-editor that is core to Bevy must both respect
+this design, and work with the limitations and abilities of Cargo.
+
+## User-facing explanation
+
+The easiest way to get started with Bevy's editor by downloading it with Cargo:
+
+```
+$ cargo install bevy
+```
+
+*Note the use of `install`, not `add`!* This means Bevy's editor for your entire rust environment. In the future you may also be able to download it from external sources such as Steam.
+
+This will add `bevy` to our path:
+
+```
+$ bevy
+Bevy's Command Line Tool
+
+Usage: bevy [OPTIONS] [COMMAND]
+
+...
+```
+
+This tool allows us to create new projects, manage assets, edit projects, and
+even publish games. Note that `bevy` is not meant to replace `cargo`, but to
+work alongside it.
+
+Also, just like Cargo, the Bevy library version used in a project does not need to match
+the version of `bevy` you have installed. You can upgrade and downgrade them
+independently. Therefore, we recommend you always use the latest version of the
+`bevy` tool.
+
+Bevy's editor can be used graphically, or as a command line tool. This document
+covers the command line usage, but running `bevy edit` in a project will open
+the graphical view.
+
+**The `bevy` command line interface is not stable. Do not use it in build
+scripts!**
+
+### Default Project Structure
+
+The first entry point for the `bevy` command is `new` and `init`, for creating
+new projects, meant to mimic the commands in cargo:
+
+```
+bevy new [options] <project-name>
+```
+
+As an example, let's create a new project, `flappers`:
+
+```
+$ bevy new flappers
+```
+
+This creates a new `flappers` folder in your current directory, using the
+default template, similar to using `cargo new`. Unlike Cargo, however, a Bevy
+project can start from any number of templates you can get from the Bevy asset
+repository.
+
+Next, let's take a look at the project structure:
+
+```
+flappers/
+    README.md
+    Cargo.toml
+    LICENSE-Apache2.txt
+    assets/
+        ...
+    src/
+        main.rs
+    tests/
+        ...
+    target/
+        ...
+    editor/
+        Cargo.toml
+        src/
+            main.rs
+    .git/
+        ...
+    .github/
+        workflows/
+            ...
+```
+
+The default template provides a lot more than a standard new cargo project. Much
+of this is not strictly required, but will be wanted for the average project.
+Because this is mostly convention, feel free to change the layout as needed. The
+first three files: `RREADME.md`, `Cargo.toml`, and `LICENSE-Apache2.txt` are
+standard meta-information that are in most crates. The next folder: `assets` is
+the default location to place project non-code assets. The bulk of the source
+code goes in the `src` directory, likewise tests go in `tests`. Project builds
+go into the `target` directory.
+
+The `editor` directory provides a sub-crate for the Bevy editor. Associating
```

Comment:
I don't understand this entire paragraph, sorry. What does `associating` mean here, how is that related with Bevy versions, and why is there a crate involved? I'm clueless. 🤔